client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,600 @@
+    setDraftSheetName(getValues(`resultTallySheets[${currentSheetIndex}].name`))
+    setIsRenamingSheet(true)
+  }
+
+  const confirmSheetRename = () => {
+    setValue(`resultTallySheets[${currentSheetIndex}].name`, draftSheetName, {
+      shouldDirty: true,
+    })
+    setIsRenamingSheet(false)
+    setAreAdditionalSheetActionsOpen(false)
+    setDraftSheetName('')
+  }
+
+  const cancelSheetRename = () => {
+    setIsRenamingSheet(false)
+    setAreAdditionalSheetActionsOpen(false)
+    setDraftSheetName('')
+  }
+
+  const removeSheet = () => {
+    const numSheetsAfterDeletion = sheetFields.length - 1
+
+    if (numSheetsAfterDeletion === 1 || currentSheetIndex === 0) {
+      selectVoteTotalsTab()
+    } else {
+      // Open the sheet before the to-be-deleted sheet
+      selectTabWithSheetIndex(Math.max(currentSheetIndex - 1, 0))
+    }
+
+    // If we drop back down to 1 sheet, rename that sheet "Sheet 1" under the hood
+    if (numSheetsAfterDeletion === 1) {
+      const remainingIndex = currentSheetIndex === 1 ? 0 : 1
+      setValue(`resultTallySheets[${remainingIndex}].name`, 'Sheet 1')
+    }
+
+    sheetFieldArrayMethods.remove(currentSheetIndex)
+    setAreAdditionalSheetActionsOpen(false)
+  }
+
+  const onValidSubmit: SubmitHandler<IBatchRoundDataEntryFormState> = async ({
+    resultTallySheets,
+  }) => {
+    if (!batch) {
+      return
+    }
+    try {
+      await saveBatchResults(resultTallySheets)
+    } catch {
+      // Errors are automatically toasted by the queryClient
+      setIsEditing(true)
+      return
+    }
+    selectVoteTotalsTab()
+    // Reset the form's isDirty value back to false
+    reset({ resultTallySheets })
+    setIsEditing(false)
+  }
+
+  const onInvalidSubmit: SubmitErrorHandler<IBatchRoundDataEntryFormState> = errors => {
+    // Open the first sheet with a validation error
+    if (sheetFields.length > 1) {
+      const sheetWithErrorIndex = (errors.resultTallySheets || []).findIndex(
+        Boolean
+      )
+      if (sheetWithErrorIndex !== -1) {
+        selectTabWithSheetIndex(sheetWithErrorIndex)
+      }
+    }
+  }
+
+  const saveResults = (e: BaseSyntheticEvent) => {
+    handleSubmit(onValidSubmit, onInvalidSubmit)(e)
+  }
+
+  const discardChanges = () => {
+    reset({ resultTallySheets: initialResultTallySheets })
+    selectVoteTotalsTab()
+    setIsEditing(false)
+  }
+
+  // -------------------- Handlers (end) --------------------
+
+  const currentSheetName = watch(`resultTallySheets[${currentSheetIndex}].name`)
+
+  const draftSheetNameError = (() => {
+    if (!isRenamingSheet) {
+      return ''
+    }
+    if (!draftSheetName) {
+      return 'Required'
+    }
+    const sheetNames = sheetFields.map((_, i) =>
+      i === currentSheetIndex
+        ? draftSheetName
+        : watch(`resultTallySheets[${i}].name`)
+    )
+    if (new Set(sheetNames).size < sheetNames.length) {
+      return 'Must be unique'
+    }
+    return ''
+  })()
```

Comment:
As you've probably started to notice, sheet name changes are handled outside of react-hook-form up until the point a new sheet name is confirmed. At that point, we call `setValue` to plug back into react-hook-form

Lots of reasons for this, the key one being that we can't always render the sheet name input (even as a hidden input) when it's tucked away inside the popover. And as you'll also see elsewhere in this PR, for react-hook-form to work ideally, inputs should always be present and not unmounting and remounting